src/test/ui/async-await/issues/issue-78600.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0760]: `async fn` return type cannot contain a projection or `Self` that references lifetimes from a parent scope
+  --> $DIR/issue-78600.rs:6:33
+   |
+LL |     async fn new(i: &'a i32) -> Result<Self, ()> {
+   |                                 ^^^^^^^----^^^^^
+   |                                        |
+   |                                        help: consider spelling out the type instead: `std::result::Result<S<'a>, ()>`
```

Comment:
How hard would it be to only print the Self part of the type, i.e. `S<'a>`?

It seems like you might be able to save it in the `ProhibitOpaqueVisitor`. Might not work in every case.